fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/ScalarOperatorFunctions.java

```diff
@@ -293,6 +294,16 @@ public static ConstantOperator floor(ConstantOperator expr) {
     /**
      * Arithmetic function
      */
+    @FEFunction(name = "add", argTypes = {"SMALLINT", "SMALLINT"}, returnType = "SMALLINT")
+    public static ConstantOperator addSmallInt(ConstantOperator first, ConstantOperator second) {
+        return ConstantOperator.createSmallInt((short) Math.addExact(first.getSmallint(), second.getSmallint()));
+    }
```

Comment:
This won't overflow/underflow, because it will only be called after type conversion with common type is SMALLINT, that means both of operand are less than SMALLINT.